docs/executor_tutorial/google_lifesciences.rst

```diff
@@ -0,0 +1,880 @@
+    Finished download.
+    [Fri Apr 17 20:53:10 2020]
+    Finished job 0.
+    9 of 9 steps (100%) done
+    Complete log: /home/vanessa/snakemake-work/tutorial/.snakemake/log/2020-04-17T202749.218820.snakemake.log
+
+
+We've finished the run, great! Let's inspect our results.
+
+Step 4: View Results
+::::::::::::::::::::
+
+The entirety of the log that was printed to the terminal will be available
+on your local machine where you submit the job in the hidden `.snakemake`
+folder under "log" and timestamped accordingly. If you look at the last line
+in the output above, you'll see the full path to this file.
+
+You also might notice a line about downloading results:
+
+.. code:: console
+
+    Downloading from remote: snakemake-testing-data/plots/quals.svg
+
+
+Since we defined this to be the target of our run
+
+.. code:: console
+
+
+    rule all:
+        input:
+            "plots/quals.svg"
+
+
+this fill is downloaded to our host too. Actually, you'll notice
+that paths in storage are mirrored on your filesystem (this is what the workers
+do too):
+
+
+.. code:: console
+
+    $ tree snakemake-testing-data/
+    snakemake-testing-data/
+    └── plots
+        └── quals.svg
+
+
+We can see the result of our run, quals.svg, below:
+
+.. image:: workflow/quals.svg
+
+
+And if we look at storage, we see that the result file (under plots) and intermediate
+results (under sorted_reads and calls) are saved there too!
+
+.. image:: workflow/results-google-storage.png
+
+The source folder contains a cache folder with archives that contain your working directories,
+one for each step. You can safely delete this folder, or keep it if you want to reproduce
+the run in the future.
+
+
+Step 5: Debugging
+:::::::::::::::::
+
+Let's introduce an error (purposefully) into our Snakefile to practice debugging.
+Let's remove the conda environment.yml file for the first rule, so we would
+expect that Snakemake won't be able to find the executables for bwa and samtools.
+In your Snakefile, change this:
+
+.. code:: python
+
+    rule bwa_map:
+        input:
+        fastq="samples/{sample}.fastq",
+        idx=multiext("genome.fa", ".amb", ".ann", ".bwt", ".pac", ".sa")
+    conda:
+        "environment.yml"
+    output:
+        "mapped_reads/{sample}.bam"
+    params:
+        idx=lambda w, input: os.path.splitext(input.idx[0])[0]
+    shell:
+        "bwa mem {params.idx} {input.fastq} | samtools view -Sb - > {output}"
+
+
+to this:
+
+.. code:: python
+
+    rule bwa_map:
+        input:
+        fastq="samples/{sample}.fastq",
+        idx=multiext("genome.fa", ".amb", ".ann", ".bwt", ".pac", ".sa")
+    output:
+        "mapped_reads/{sample}.bam"
+    params:
+        idx=lambda w, input: os.path.splitext(input.idx[0])[0]
+    shell:
+        "bwa mem {params.idx} {input.fastq} | samtools view -Sb - > {output}"
+
+
+And then you'll need to remove the plots, mapped_reads, and calls folders
+so the step will run again. Then, run the job again:
```

Comment:
It is easier to tell people here that they can force re-execution of the entire workflow with --forceall.

Reply:
All set.